ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/MLSdkAsyncHttpResponseHandler.java

```diff
@@ -197,27 +148,44 @@ public void onError(Throwable t) {
                     t instanceof NullPointerException ? "NullPointerException" : t.getMessage(),
                     t
                 );
-            response(tensorOutputs);
+            response();
         }
 
         @Override
         public void onComplete() {
-            response(tensorOutputs);
+            response();
         }
     }
 
-    private void response(Map<Integer, ModelTensors> tensors) {
-        processResponse(statusCode, responseBody.toString(), parameters, tensorOutputs);
-        executionContext.getCountDownLatch().countDown();
-        // when countdown's count equals to 0 means all responses are received.
-        if (executionContext.getCountDownLatch().getCount() == 0) {
-            if (executionContext.getExceptionHolder().get() != null) {
-                actionListener.onFailure(executionContext.getExceptionHolder().get());
-                return;
+    private void response() {
```

Comment:
In this method there're three places are dealing with exception with similar codes, and the `listener.onFailure` is not near them, suggest adding a new method to deal with the exceptions like this:
```
void handleException(Exception e) {
  if (exceptionHolder.get() != null) {
    actionListener.onFailure(exceptionHolder.get());
  } else {
    actionListener.onFailure(e);
  }
}
```

Reply:
ack